.github/workflows/deploy.yml

```diff
@@ -0,0 +1,16 @@
+name: deploy
+
+on:
+  push:
+    branches:
+      - master
```

Comment:
I think in this case job will deploy concurrently with the test. Will that be a issue?